### ISSUE TITLE
Write to APEX_DEBUG.message

### ISSUE THREAD

**Issue by jeffreykemp** [OraOpenSource/Logger#158]
When debugging an Apex application one often uses the View Debug feature in Apex. It would be nice to have all logger logs duplicated in the Apex debug log as well - that way you can see your debug logs in context with the Apex logs.

```
APEX_DEBUG.message
  (p_message => p_scope || ': ' || p_text
  ,p_level   => map from logger level to APEX_ERROR.c_log_level_error / c_log_level_warn / c_log_level_info
  );
```

Maybe make it configurable so that only certain log levels are copied to the Apex debug log.

**Comment by martindsouza** [comment]
Would expanding the plugins for all the various levels help? https://github.com/OraOpenSource/Logger/blob/master/docs/Plugins.md

**Comment by jeffreykemp** [comment]
Maybe - but I would be concerned about performance if we put a plugin that has to get evaluated and executed for each and every call to logger.

**Comment by martindsouza** [comment]
@jeffreykemp Agreed. One thing to remember is that we only dynamically call the plugin code if a plugin has been registered. I.e. you could do this in dev but not register a plugin in prod and there should not be a performance hit then. There's a conditional compilation flag that turns it on/off (at the moment on for errors).

If not, how do you suggest we add in a call to the `apex_debug` package? How would we turn on/off this feature?

**Comment by jeffreykemp** [comment]
Calls to apex_debug are controlled by apex - so it should be safe to just call it always; Apex decides whether or not to record the log or not based on the current session and instance settings.

**Comment by martindsouza** [comment]
Sounds fair. Here's what I think needs to be done:

- [ ] Determine performance impact of making calls to apex_debug when:
-- [ ] Not in an apex session
-- [ ] In an apex session but no debugging

If no performance impacts then we'd need to do the following:

- [ ] Modify internal logger insert to also call `apex_debug`
- [ ] Make the changed code conditional compiled based on the presence of the `apex_debug.message` package.


I have tagged this as a Future item as the next build has a lot of items in it.

**Comment by jeffreykemp** [comment]
👍
